tests/integration/services/user.test.ts

```diff
@@ -442,6 +442,59 @@ describe('user service', () => {
             const userUpdated = await UserService.edit(address, data);
 
             expect(userUpdated).to.include(data);
+        })
+    });
+
+    describe('notifications', () => {
+        let users: User[];
+
+        before(async () => {
+            users = await UserFactory({ n: 2 });
+            const yesterday = new Date();
+            yesterday.setDate(yesterday.getDate() - 1);
+            await models.appNotification.bulkCreate([
+                {
+                    address: users[0].address,
+                    type: 1,
+                    params: {
+                        communityId: 1
+                    },
+                }, {
+                    address: users[0].address,
+                    type: 2,
+                    params: {
+                        communityId: 1
+                    },
+                }, {
+                    address: users[1].address,
+                    type: 1,
+                    params: {
+                        communityId: 2
+                    },
+                }
+            ]);
+        });
+
+        it('get all notifications from a user', async () => {
+            const notifications = await UserService.getNotifications(users[0].address, {
+                limit: '10',
+                offset: '0',
+            });
+
+            expect(notifications.length).to.be.equal(2);
+            notifications.forEach(notification => {
+                expect(notification.read).to.be.false;
+            });
+        });
+
+        it('mark all notifications as read', async () => {
+            await UserService.readNotifications(users[0].address);
+
+            const firstUser = await UserService.getUnreadNotifications(users[0].address);
+            const secondUser = await UserService.getUnreadNotifications(users[1].address);
+
+            expect(firstUser).to.be.equal(0);
+            expect(secondUser).to.be.equal(1);
```

Comment:
I see these tests are dependant on each other. For example, if they are reordered, one of them will fail. It's good practice not to do it.

Reply:
I don't know if a understood well. No matter the order, always the users[0] will have 2 notifications (that, in the second test, will be read), and the users[1] always have one notification (that wouldn't be read), because in the `before` I created the notifications based on the user's address.

Reply:
If you run 'mark all notifications as read' before 'get all notifications from a user', it will fail in 'get all notifications from a user' because 'mark all notifications as read' changes all the `users[0]` notifications to read and will fail at line 486, because `read` will be true. Or does `before` run each time before an `it` test. I don't think so. I think it's `beforeEach`. But you don't need to do that. Instead, you can create 3 users and make tests independent.

Reply:
got it!
you are right, the `before` runs once before the tests.
I followed your suggestion and used 3 users